ethcore/src/miner/miner.rs

```diff
@@ -680,6 +698,12 @@ impl Miner {
 						Err(e)
 					},
 					Ok(transaction) => {
+						// This check goes here because verify_transaction takes SignedTransaction parameter
+						if self.engine.machine().verify_transaction(&transaction, &best_block_header, client.as_block_chain_client()).is_err() {
```

Comment:
The error here should not be interpreted as permission failure. Other checks might be added in the future. Best would just to pass it to the caller:
```rust
self.engine.machine().verify_transaction(&transaction, &best_block_header, client.as_block_chain_client())?;
```